inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/v2/SortClusterConfigHolder.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.sort.standalone.config.holder.v2;
+
+import org.apache.inlong.common.pojo.sort.SortConfig;
+import org.apache.inlong.common.pojo.sort.SortTaskConfig;
+import org.apache.inlong.sort.standalone.config.holder.CommonPropertiesHolder;
+import org.apache.inlong.sort.standalone.config.holder.SortClusterConfigType;
+import org.apache.inlong.sort.standalone.config.loader.v2.ClassResourceSortClusterConfigLoader;
+import org.apache.inlong.sort.standalone.config.loader.v2.ManagerSortClusterConfigLoader;
+import org.apache.inlong.sort.standalone.config.loader.v2.SortConfigLoader;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.ClassUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.flume.Context;
+
+import java.util.Date;
+import java.util.Timer;
+import java.util.TimerTask;
+
+import static org.apache.inlong.sort.standalone.utils.Constants.RELOAD_INTERVAL;
+
+@Slf4j
+public class SortClusterConfigHolder {
+
+    private static SortClusterConfigHolder instance;
+
+    private long reloadInterval;
+    private Timer reloadTimer;
+    private SortConfigLoader loader;
+    private SortConfig config;
+
+    private SortClusterConfigHolder() {
+
+    }
+
+    private static SortClusterConfigHolder get() {
+        if (instance != null) {
+            return instance;
+        }
+
+        synchronized (SortClusterConfigHolder.class) {
+            instance = new SortClusterConfigHolder();
```

Comment:
null check after `synchronized` is needed for singleton

Reply:
fixed, thx